engine/src/main/java/com/datatorrent/stram/StramClient.java

```diff
@@ -572,6 +572,8 @@ public void startApplication() throws YarnException, IOException
       vargs.add("-Dhadoop.root.logger=" + (dag.isDebug() ? "DEBUG" : "INFO") + ",RFA");
       vargs.add("-Dhadoop.log.dir=" + ApplicationConstants.LOG_DIR_EXPANSION_VAR);
       vargs.add(String.format("-D%s=%s", StreamingContainer.PROP_APP_PATH, dag.assertAppPath()));
+      vargs.add(String.format("-D%s=$'%s'", LogicalPlan.APPLICATION_NAME.getLongName(),
```

Comment:
To avoid code duplication, introduce static helper method in StramClientUtils.java.